pandas/tests/io/json/test_normalize.py

```diff
@@ -294,14 +294,59 @@ def test_missing_field(self, author_missing_data):
         expected = DataFrame(ex_data)
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize("max_level", [0, 1])
+    def test_max_level_with_records_path(self,
+                                         max_level):
+        # GH23843: Enhanced JSON normalize
+        test_input = [{'CreatedBy': {'Name': 'User001'},
+                       'Lookup': [{'TextField': 'Some text',
+                                   'UserField': {'Id': 'ID001',
+                                                 'Name': 'Name001'}},
+                                  {'TextField': 'Some text',
+                                   'UserField': {'Id': 'ID001',
+                                                 'Name': 'Name001'}}
+                                  ],
+                       'Image': {'a': 'b'},
+                       'tags': [{'foo': 'something', 'bar': 'else'},
+                                {'foo': 'something2', 'bar': 'else2'}]
+                       }]
+
+        expected_data = {0: [{"TextField": "Some text",
+                              'UserField': {'Id': 'ID001',
+                                            'Name': 'Name001'},
+                              "CreatedBy": {"Name": "User001"},
+                              'Image': {'a': 'b'}},
+                             {"TextField": "Some text",
+                              'UserField': {'Id': 'ID001',
+                                            'Name': 'Name001'},
+                              "CreatedBy": {"Name": "User001"},
+                              'Image': {'a': 'b'}}],
+                         1: [{"TextField": "Some text",
+                              "UserField.Id": "ID001",
+                              "UserField.Name": "Name001",
+                              "CreatedBy": {"Name": "User001"},
+                              'Image': {'a': 'b'}},
+                             {"TextField": "Some text",
+                              "UserField.Id": "ID001",
+                              "UserField.Name": "Name001",
+                              "CreatedBy": {"Name": "User001"},
+                              'Image': {'a': 'b'}}]}
+        expected_data = expected_data[max_level]
```

Comment:
Same comment as below - typically better to pass the expected value in via parametrization rather than using a local dict